### PR TITLE
Cicd

### PR DESCRIPTION
**Pull request by marcinszremski**
Pipeline for deployment and release has been added. 3 Stages : dev, qa, staging

### CODE REVIEW

File: .github/workflows/pr-pipeline.yaml [hunk 1]
```diff
@@ -4,52 +4,48 @@ on:
...
-        EODH_AUTH__USERNAME: ${{ secrets.EODH_AUTH__USERNAME }}
-        EODH_AUTH__PASSWORD: ${{ secrets.EODH_AUTH__PASSWORD }}
-        EODH_AUTH__CLIENT_ID: ${{ secrets.EODH_AUTH__CLIENT_ID }}
-        EODH_AUTH__CLIENT_SECRET: ${{ secrets.EODH_AUTH__CLIENT_SECRET }}
-
-    - name: Get code coverage
-      uses: orgoro/coverage@v3.2
-      with:
+      - name: Check out the repository
+        uses: actions/checkout@v2
```
Comment (xultaeculcis) [inline .github/workflows/pr-pipeline.yaml:18]: Can we bump the checkout action to `v4`

File: .github/workflows/build-and-deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,129 @@
+name: build-and-deploy
+
+on:
+  push:
+    branches:
+      - main
+permissions: write-all
```
Comment (xultaeculcis) [inline .github/workflows/build-and-deploy.yaml:7]: Since you are specifying permissions on per-job basis this statement is redundant

**Review by xultaeculcis — APPROVED** [review]
LGTM. Few comments but nothing major. Treat them as suggestions ;)

File: .github/workflows/build-and-deploy.yaml (hunk 2 above)
Comment (marcinszremski) [reply .github/workflows/build-and-deploy.yaml]: Updated

File: .github/workflows/build-and-deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+jobs:
+  build-push-release-dev:
+    runs-on: ubuntu-latest
+    environment: dev
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (marcinszremski) [reply .github/workflows/build-and-deploy.yaml]: Updated

**Pull request merged**